### ISSUE TITLE
Wagtail Brand Font

### ISSUE THREAD

**Issue by vsalvino** [wagtail/wagtail#7724]
The issue of font choice recently came up in https://github.com/wagtail/sphinx_wagtail_theme/issues/120

@allcaps mentioned in a discussion on slack about Wagtail finding a font for use in docs, and elsewhere, that is distinctly Wagtail. Generally speaking I think providing reusable resources of any kind is a net positive for helping strengthen the Wagtail ecosystem.

Currently there is a mix on fonts in use:
* Wagtail Admin: Open Sans
* Wagtail website and docs: Apercu (this aligns with Torchbox branding)
* `sphinx_wagtail_theme` package has recently switched to Source Sans due to legal reasons, as a stop-gap measure until a more suitable replacement is found.

The issue referenced above means we cannot legally distribute Apercu with the Wagtail docs pip package, nor can the Apercu font files legally be committed on GitHub.

With the new Wagtail admin design refresh on the horizon, it seems like an opportunity to make font usage (and other branding elements such as colors) consistent between the various parts of the Wagtail ecosystem.

I personally have no opinion in the matter, other than consistency. Here are a couple options:
* Wagtail community as a whole centers on an open source font.
* Torchbox (or other sponsor) purchases a special font license that permits redistribution within the confines of the Wagtail ecosystem. Perhaps Colophon (maker of Apercu) or a smaller boutique font foundry would be willing to make special dispensation (for a fee)?
* Volunteer or paid consultant makes a custom font for the Wagtail org. Usually such an endeavor starts as a fork of a robust open source font and tweaks the glyphs (e.g. start with Source Sans and modify the glyphs to look wagtail-ish).

**Comment by thibaudcolas** [comment]
Quick update from the [page editor design refresh](https://github.com/wagtail/wagtail/discussions/7739) mentioned by Vince:

- We most likely won’t retain Open Sans
- We’re trying out either sans-serif fonts available on Google Fonts, or a system font (San Francisco on macOS & iOS, Roboto on Android, Segoe UI on Windows)

Last time we spoke @benenright and I were having trouble figuring out how to get our hands on a Segoe UI font file for macOS, so we can test the designs with it – If anyone else knows more about this please let us know.

**Comment by Scotchester** [comment]
+1 vote for using system fonts. I wouldn't worry a ton about Segoe; in my experience the metrics are close enough to San Franscisco that it seems unlikely adjustments would be needed. That said, there are plenty of folks around who could help test them out in Windows and relay screenshots or video.

**Comment by Scotchester** [comment]
I will also note that @tomdyson also commented in favor of system fonts, either in Slack or in a Core Team meeting.

**Comment by vsalvino** [comment]
For the wagtail admin itself, I am also in favor of system fonts. Reason being that the admin is already quite heavy and we can shave off several megabytes of payload. We have already implemented this in coderedcms, as an effort to speed up the admin (because our streamfields are so big).

@thibaudcolas this might be helpful: https://developer.microsoft.com/en-us/fluentui#/resources. BUT, I would caution that testing segoe on macos will likely be fruitless. Windows has a totally different way of handling font rendering, DPI, screen sizes, etc. The mac has very smooth/thick font rendering, and very small UI elements by default. Windows on the other hand has much weaker font rendering, and the UI elements are much larger by default (due to display scaling - most laptops default to 150% scaling). The end result is that things like input controls could end up looking very unnatural in Windows if they have lots of padding and a light weight font (300). Whereas this might look "normal" on a mac.

For an example implementation of native font stack, look to Bootstrap as they have a well-tested implementation: https://getbootstrap.com/docs/5.1/content/reboot/

**Comment by thibaudcolas** [comment]
Thank you both! Re testing, this would be in Figma at this stage. I believe it implements its own font rendering but I don’t know how it compares with the rendering of Windows and macOS.

@vsalvino do you have more info on how Bootstrap picked their font stack? The implementation I’ve referred to in the past has been GitHub’s (https://markdotto.com/2018/02/07/github-system-fonts/):

```
-apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji"
```

Their article is a few years old but on there they document issues with `system-ui` and Linux system fonts.

**Comment by nmorduch** [comment]
I agree with following Bootstrap's approach: https://getbootstrap.com/docs/5.1/content/reboot/#native-font-stack. I was also looking at similar references in the docs of [Any Design](https://ant.design/docs/spec/font) though I don't remember how I got there (maybe from a search engine?)

I was worried about overriding fonts, for example if people use the Dyslexie typeface, so I checked out their [Chrome extension](https://chrome.google.com/webstore/detail/dyslexie-font-dyslexia-ac/jphdkcjkapjfojcjajeeomimmihnghjp) and it does a fine job of overriding font choices except if it's trying to override `!important` styles, so we should be extra sure to avoid using `!important`.

My other concern with a font stack like this is font weights. Not only because, as @vsalvino noted, fonts show lighter on Windows, but also because not every font is going to have a lot of weights available. So we should make sure that anytime we use font weight to convey semantic information, that does not get lost using a font like Dyslexie (which, as far as I can tell, has only two weights).

**Comment by nmorduch** [comment]
I just noticed that [uswds also has a system font option](https://designsystem.digital.gov/design-tokens/typesetting/font-family/), which looks pretty similar to the others.

**Comment by thibaudcolas** [comment]
👍 here are the options floated so far side-by-side for comparison – Bootstrap, GitHub, Ant Design, USWDS

```
system-ui, -apple-system, "Segoe UI", Roboto, "Helvetica Neue", Arial, "Noto Sans", "Liberation Sans", sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
-apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji"
-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, 'Helvetica Neue', Arial, 'Noto Sans', sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol"
```

@nmorduch do you have more info on the weights of the Dyslexie font? This feels like something we might want to explicitly support, even if that just means being aware of it / documenting how we can test it.

**Comment by thibaudcolas** [comment]
We have people on Slack volunteering to help! Message: https://wagtailcms.slack.com/archives/C014L7KJH3N/p1644402583455259.

I would like us to trial the following options:

```scss
// Bootstrap
system-ui, -apple-system, "Segoe UI", Roboto, "Helvetica Neue", Arial, "Noto Sans", "Liberation Sans", sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji"
// GitHub
-apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji"
// Ant Design
-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial, "Noto Sans", sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji"
// USWDS
-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol"

// GitHub with system-ui after Segoe UI
-apple-system, BlinkMacSystemFont, "Segoe UI", system-ui, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji"
// Ant Design with system-ui after Segoe UI
-apple-system, BlinkMacSystemFont, "Segoe UI", system-ui, Roboto, "Helvetica Neue", Arial, "Noto Sans", sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji"
// USWDS with system-ui after Segoe UI
-apple-system, BlinkMacSystemFont, "Segoe UI", system-ui, Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol"
```

**Comment by Scotchester** [comment]
> Windows machines that don’t have Segoe UI – apparently that’s Windows XP, Vista, 7, and potentially some versions of Windows Server

Segoe UI became the Windows system font beginning in Vista (and Server 2008). See [its Wikipedia entry](https://en.wikipedia.org/wiki/Segoe) (first paragraph) and [Microsoft documentation](https://docs.microsoft.com/en-us/typography/font-list/segoe-ui#products-that-supply-this-font).

**Comment by vsalvino** [comment]
Has the new font stack also been tested on every supported language in the Wagtail admin? I'm mainly thinking non-Latin alphabets (Arabic, Asian scripts). In those cases some font substitutions will probably be in use, so want to make sure the stack looks OK.

**Comment by thibaudcolas** [comment]
I believe the only script we’ve tested it on that’s not english is Simplified Chinese, since that’s what the system-ui issues were reported on. It’d be nice to know how Wagtail works with other scripts!
